### DISCUSSION TITLE
Jellyfin server(s) as music source

### DISCUSSION THREAD

**Discussion by ohhai-kthxbai** [music-assistant/support#407] [Feature requests and ideas]
Please could you add jellyfin server as a music source.

## JELLY FIN IS AVAILABLE https://music-assistant.io/music-providers/jellyfin/

**Comment by marcelveldt** [comment]
Tbh I'd rather invest my time in implementing the missing features then let MA connect to all existing similar products out there.
I think that if Jellyfin currently suits your needs you should stick with that for now. Of course I'm open for PR's if somebody want to implement the music provider plugin for MA. It's just that I will not likely invest my time in that unless this request gets a very high amount of user votes and there actually is a valid usecase I'm currently not seeing.

**Comment by linus-ha** [comment]
Accepting DLNA in general as a music source would allow PLEX, Jellyfin, Emby, Logitech Media Server, Kodi and other to be a source for music. I don't think that individual source implementations are needed. @marcelveldt 
Doing that on the other side would mean, that direct network access is needed, for receiving DLNA advertisements.
You then may need to play them locally and redistribute them. (I don't really try understanding the architecture of MA yet, so this assumption may be wrong).

**Reply by marcelveldt** [reply]
DLNA could be a route to address this. In that case MA has a DLNA Music provider which reads the metadata from all dlna sources.
MA itself is responsible for redistributing the music streams, same as it is now for the existing music sources.

The whole architecture of MA is pluggable so all that is required is someone writing a "music provider" in the library:

https://github.com/music-assistant/music-assistant-lib/tree/master/music_assistant/music_providers

Really that is all that is required. One file (or package) that implements the methods described in the MusicProvider model.

Still I do not follow why you want to use those "competitors" as source for MA as all of them (except for LMS) are focused on Video instead of audio. So maybe see it the other way around...

**Reply by geiseri** [reply]
DLNA would solve all my issues to get music from my NAS without needing to play games with samba mounts.

**Comment by DutchJaFO** [comment]
PLEX does all 3 (photo, video and music) 

The one thing that PLEX does for music that I haven't seen anywhere else are magically generated playlists.
In addition to artist / album you also get decade, style and mood. There's also a 'top albums of [random? year]' set.

I suspect MA could do the style & decade variants (that's pure meta data ...).
Tracking statistics of music played would allow for a 'top 40 of this week' style list for your own system. Spotify does something similar by automagically generating 'best of [last year]' lists every once in a while.

Not sure how they do the 'mood' sorting, but I think it's based on analysis of the tracks themselves.

**Reply by marcelveldt** [reply]
mood is also metadata. In fact this info is already being gathered in the MA database as well as playback statistics.
My endgoal would be to create dynamic playlists where you can define some parameters yourself and let the playlist be dynamically created. It is a wish I have in mind and the whole reason I once started MA. Products I used before like Roon, LMS, Plex all came close but just that little personal tweaking I want/search is not there but maybe I'm just really demanding. Anyways... Everything that is done now is just the "plumbing" for that higher goal: To be able to really create dynamic playlists, maybe even with some AI/ML in the mix.

In the meanwhile I have no problem in creating a generic DLNA music provider if that is an easy one to do and it fulfills a temporary patch but as you can read above the real goal is to be able to put some development time for that higher goal.

**Comment by ComputerComa** [comment]
I too would agree that DLNA would be a good item to have working with this. I was going to use this as an alternative to the built in Media browser since I haven't been able to find a way to shuffle or even play playlists in the built in media browser, so I was kinda bummed when there was no DLNA support.

**Comment by beankylla** [comment]
I can see jellyfin available as a source in the documentation.
https://music-assistant.io/music-providers/jellyfin/

does this mean that this discussion can be closed ? :-)

Totally trying this out tonight 👍👍👍

**Comment by beankylla** [comment]
Small feedback from my use here: i have a pretty big music collection. 
The integration of jellyfin works fine once the music has been scanned but i've had it installed for 2,5 weeks now and there is a random number of songs / albums being added everyday, and no visibility on the progress.
Not sure if i can provide elements to help fix somethig or if this is expected? :)

**Reply by OzGav** [reply]
Are you trying with beta142?

**Comment by beankylla** [comment]
i have server version 2.0.0b144. Happy to provide logfiles.
On one of the updates  recently,  all of the library was erased so the sync restarted from scratch. 
it is not quicker :)

**Reply by OzGav** [reply]
There were some improvements in the latest betas so update to b147 but there are also still some open issues

**Reply by beankylla** [reply]
Just updated. tells me b148.
Scan of albums is incredibly quicker. that is niiice. Scan of songs does not seem to update though. I usually go to albums so don't mind ;)
there seems to be 50 albums missing vs my jellyfin setup ^^'.
Not sure where that comes from!
Anything i can do to help?

**Reply by beankylla** [reply]
is there a list of the open issues?

**Reply by OzGav** [reply]
You are on an old beta. All issues are here https://github.com/music-assistant/hass-music-assistant/issues
